pkg/manager/member/pd_failover.go

```diff
@@ -271,6 +254,78 @@ func (f *pdFailover) isPDInQuorum(tc *v1alpha1.TidbCluster) (bool, int) {
 	return healthCount > (len(tc.Status.PD.Members)+len(tc.Status.PD.PeerMembers))/2, healthCount
 }
 
+// pdFailureMemberAccess implements the FailureObjectAccess interface for PD member
+type pdFailureMemberAccess struct{}
+
+func (fma *pdFailureMemberAccess) GetMemberType() v1alpha1.MemberType {
+	return v1alpha1.PDMemberType
+}
+
+// GetFailureObjects returns the set of failure pd names
+func (fma *pdFailureMemberAccess) GetFailureObjects(tc *v1alpha1.TidbCluster) map[string]v1alpha1.EmptyStruct {
+	failurePdMembers := make(map[string]v1alpha1.EmptyStruct, len(tc.Status.PD.FailureMembers))
+	for pdName := range tc.Status.PD.FailureMembers {
+		failurePdMembers[pdName] = v1alpha1.EmptyStruct{}
+	}
+	return failurePdMembers
+}
+
+// IsFailing returns if the particular member is in unhealthy state
+func (fma *pdFailureMemberAccess) IsFailing(tc *v1alpha1.TidbCluster, pdName string) bool {
+	return !tc.Status.PD.Members[pdName].Health
+}
+
+// GetPodName returns the pod name of the given failure member
+func (fma *pdFailureMemberAccess) GetPodName(_ *v1alpha1.TidbCluster, pdName string) string {
+	return strings.Split(pdName, ".")[0]
```

Comment:
PD ID will change after scale-in and then scale-out.

Reply:
I see. Should we have some tests on this? My concern is if we scale-in pd and then scale-out back, if its `pdName` changes, then we could ultimately remove the wrong pod. What do you think? @lalitkfk @csuzhangxc

Reply:
Hi @hanlins Nothing has changed in pdFailover w.r.t. to scale-in and scale-out during failover. So this is already tested code. The only change in pdFailover is that it tries to detect HostDown(and does a pod restart) and then does the scale-in if the pod still fails to come up. Also, the wrong pdName will not be removed because the pod will only be removed if both pod and member is currently in failed/unhealthy state and the member is present in FailureMember, so there is double checking there. I have also tested multiple scenarios w.r.t pd failover as listed in the description.

Reply:
I see, thanks for the clarification.